src/MediaWiki/Search/Search.php

```diff
@@ -110,13 +118,30 @@ public function setDB( DatabaseBase $connection ) {
 		$this->fallbackSearch = null;
 	}
 
+
+	/**
+	 * @return \LoadBalancer
+	 */
+	protected function getLoadBalancer() {
+
```

Comment:
Ok, no problem.

What about [`getDefaultSearchEngineTypeForDB`](https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/1916/files?diff=unified#diff-2d41b72f98fa4903663fb448b947b11dR76)?
Not sure if I should register a callback or just add the method to `ApplicationFactory` as is.